src/client/pps.go

```diff
@@ -402,15 +402,29 @@ func (c APIClient) RestartDatum(jobID string, datumFilter []string) error {
 	return grpcutil.ScrubGRPC(err)
 }
 
-// ListDatum returns info about all datums in a Job
-func (c APIClient) ListDatum(jobID string, pageSize int64, page int64) (*pps.ListDatumResponse, error) {
+// ListDatum returns info about datums in a Job
+func (c APIClient) ListDatum(jobID string, pageSize, page int64) (*pps.ListDatumResponse, error) {
+	return c.listDatum(jobID, nil, pageSize, page)
+}
+
+// ListDatumInput returns info about datums for a pipeline with input. The
+// pipeline doesn't need to exist.
+func (c APIClient) ListDatumInput(input *pps.Input, pageSize, page int64) (*pps.ListDatumResponse, error) {
+	return c.listDatum("", input, pageSize, page)
+}
+
+func (c APIClient) listDatum(jobID string, input *pps.Input, pageSize, page int64) (*pps.ListDatumResponse, error) {
+	req := &pps.ListDatumRequest{
+		Input:    input,
+		PageSize: pageSize,
+		Page:     page,
+	}
+	if jobID != "" {
+		req.Job = NewJob(jobID)
+	}
```

Comment:
nit: you could get rid of this `if` and `req` by having the function take a `job *pps.Job` instead of a `jobID string`